src/test/java/htsjdk/samtools/util/BlockCompressedInputStreamTest.java

```diff
@@ -93,80 +93,85 @@ public void available_should_return_number_of_bytes_left_in_current_block() thro
+        return linesWritten;
+    }
+
+    @DataProvider(name = "customInflaterInput")
+    public Object[][] customInflateInput() throws IOException {
+        final File tempFile = File.createTempFile("testCustomInflater.", ".bam");
+        tempFile.deleteOnExit();
+        final List<String> linesWritten = writeTempBlockCompressedFileForInflaterTest(tempFile);
 
-        BlockGunzipper.setDefaultInflaterFactory(myInflaterFactory);
+        final InflaterFactory countingInflaterFactory = new CountingInflaterFactory();
+        BlockGunzipper.setDefaultInflaterFactory(countingInflaterFactory);
 
         return new Object[][]{
                 // use default InflaterFactory
-                {new BlockCompressedInputStream(new FileInputStream(tempFile), false), linesWritten, 4},
-                {new BlockCompressedInputStream(tempFile), linesWritten, 4},
-                {new AsyncBlockCompressedInputStream(tempFile), linesWritten, 4},
-                {new BlockCompressedInputStream(new URL("http://broadinstitute.github.io/picard/testdata/index_test.bam")), null, 21},
+                {(CheckedExceptionInputStreamSupplier) () -> new BlockCompressedInputStream(new FileInputStream(tempFile), false), linesWritten, 4},
+                {(CheckedExceptionInputStreamSupplier) () -> new BlockCompressedInputStream(tempFile), linesWritten, 4},
+                {(CheckedExceptionInputStreamSupplier) () -> new AsyncBlockCompressedInputStream(tempFile), linesWritten, 4},
+                {(CheckedExceptionInputStreamSupplier) () -> new BlockCompressedInputStream(new URL("http://broadinstitute.github.io/picard/testdata/index_test.bam")), null, 21},
                 // provide InflaterFactory
-                {new BlockCompressedInputStream(new FileInputStream(tempFile), false, myInflaterFactory), linesWritten, 4},
-                {new BlockCompressedInputStream(tempFile, myInflaterFactory), linesWritten, 4},
-                {new AsyncBlockCompressedInputStream(tempFile, myInflaterFactory), linesWritten, 4},
-                {new BlockCompressedInputStream(new URL("http://broadinstitute.github.io/picard/testdata/index_test.bam"), myInflaterFactory), null, 21}
+                {(CheckedExceptionInputStreamSupplier) () -> new BlockCompressedInputStream(new FileInputStream(tempFile), false, countingInflaterFactory), linesWritten, 4},
+                {(CheckedExceptionInputStreamSupplier) () -> new BlockCompressedInputStream(tempFile, countingInflaterFactory), linesWritten, 4},
+                {(CheckedExceptionInputStreamSupplier) () -> new AsyncBlockCompressedInputStream(tempFile, countingInflaterFactory), linesWritten, 4},
+                {(CheckedExceptionInputStreamSupplier) () -> new BlockCompressedInputStream(new URL("http://broadinstitute.github.io/picard/testdata/index_test.bam"), countingInflaterFactory), null, 21}
         };
     }
 
     @Test(dataProvider = "customInflaterInput")
```

Comment:
Could you make this `@Test(dataProvider = "customInflaterInput" singleThreaded = true)` since it's known to be not safe.